src/core/qgsbookmarkmodel.cpp

```diff
@@ -236,14 +234,11 @@ bool QgsBookmarkManagerModel::insertRows( int, int count, const QModelIndex &par
     mBlocked = false;
     result &= res;
   }
-  endInsertRows();
   return result;
 }
 
-bool QgsBookmarkManagerModel::removeRows( int row, int count, const QModelIndex &parent )
+bool QgsBookmarkManagerModel::removeRows( int row, int count, const QModelIndex & )
 {
-  beginRemoveRows( parent, row, row + count );
```

Comment:
I don't understand this change, doesn't it break the model?

Reply:
On the contrary it fixes it: beginRemoveRows and all other mandatory signals are already called by 
https://github.com/qgis/QGIS/blob/master/src/core/qgsbookmarkmodel.cpp#L29
https://github.com/qgis/QGIS/blob/master/src/core/qgsbookmarkmanager.cpp#L169
https://github.com/qgis/QGIS/blob/master/src/core/qgsbookmarkmodel.cpp#L294
https://github.com/qgis/QGIS/blob/master/src/core/qgsbookmarkmanager.cpp#L169

The issue here was that beingRemoveRows was called twice before calling (twice again) endRemoveRows, the same happened to the other signals.

Reply:
Ah, fair enough!